src/TreeSupport.h

```diff
@@ -59,6 +59,12 @@ class TreeSupport
          , parent(parent)
         {}
 
+        ~Node()
+        {
+            parent = nullptr;
+            merged_neighbours.clear();
```

Comment:
This has no effect. The instance is about to be deleted and this memory is about to be unallocated, so whether the unallocated memory says 0 or not makes no difference.

Reply:
If there's no potential segfault, this has no effect, but this leave a trace of clue if a segfault happens. If a deleted object is picked up later (which shouldn't happen), it may not lead to a direct segfault immediately, but when you try to access something inside, it can. Like `node->parent` works (by chances), but `node->parent->parent` doesn't (segfault by chance.)

I added this reset code here is to "cleanup" the deleted object which shouldn't be accessed later. If in case something happens and we are debugging at this location, this "empty" node object can give us a good clue of what might have happened. Otherwise, if we pick up a random Node with old data in it while we shouldn't, it's impossible to tell at the first glance because that Node contains a valid old data.

Reply:
I think we should work from the assumption that no segfault occurs though.

Especially since this is live code that is run during tree support. Zeroing all that memory is just a waste of processing power for our users. If you really need this we should at least put this between `#ifdef DEBUG` statements and document why we do this.

Reply:
As discussed, I put that into a DEBUG block so it won't affect release builds. Commit: ac1fcea